runtime/binding-mqtt/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/internal/stream/MqttClientFactory.java

```diff
@@ -2295,16 +2296,14 @@ private void doEncodeConnect(
                     .keepAlive((int) MILLISECONDS.toSeconds(keepAliveMillis))
                     .build();
 
-            doNetworkData(traceId, authorization, 0L, connectHeader);
-
             final MqttConnectV5FW connect =
-                mqttConnectV5RW.wrap(writeBuffer, FIELD_OFFSET_PAYLOAD, writeBuffer.capacity())
+                mqttConnectV5RW.wrap(writeBuffer, connectHeader.limit(), writeBuffer.capacity())
```

Comment:
Please use extends in `protocol.idl` so that `MqttConnectV5` can be created here using a single flyweight.